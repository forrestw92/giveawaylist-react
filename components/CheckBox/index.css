.checkbox {
  margin-right: 10px;
  appearance: none;
  display: flex;
  -webkit-appearance: none;
  width: 16px;
  height: 16px;
  border: 1px solid rgba(0, 0, 0, 0.3);
  border-radius: 0;
  background: inherit;
}
.checkbox:checked:after {
  content: "";
  background: #f28a2c;
  height: 8px;
  width: 8px;
  transition: all 0.33s ease;
  margin: auto;
}
.label {
  margin-bottom: 10px;
  display: flex;
  align-items: center;
}
