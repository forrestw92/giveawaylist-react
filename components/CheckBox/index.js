import React from "react";
import { string, func } from "prop-types";
import stylesheet from "./index.css";

class CheckBox extends React.Component {
  render() {
    const { id, label, name, _onChange } = this.props;
    return (
      <label htmlFor={id}>
        <input
          type={"checkbox"}
          name={name}
          id={id}
          className={stylesheet["checkbox"]}
          onChange={e => _onChange(e, name)}
        />
        {label}
      </label>
    );
  }
}
CheckBox.propTypes = {
  id: string.isRequired,
  name: string.isRequired,
  label: string.isRequired,
  _onChange: func.isRequired
};

export default CheckBox;
