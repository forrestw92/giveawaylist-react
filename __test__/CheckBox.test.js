/* eslint-env jest */

import { shallow, render } from "enzyme";
import React from "react";
import renderer from "react-test-renderer";
import Router from "next/router";
import CheckBox from "../components/CheckBox";
const mockedRouter = { push: () => {}, prefetch: () => {} };
Router.router = mockedRouter;

describe("CheckBox Component", () => {
  const appShallow = shallow(
    <CheckBox
      _onChange={() => {}}
      label={"Hide Books"}
      name={"hideBooks"}
      id={"hideBooks"}
    />
  );
  const appRender = render(
    <CheckBox
      _onChange={() => {}}
      label={"Hide Books"}
      name={"hideBooks"}
      id={"hideBooks"}
    />
  );
  it("has Label", () => {
    expect(appShallow.find("label").text()).toEqual("Hide Books");
  });
  it("has Checkbox", () => {
    expect(appShallow.find("input").hasClass("checkbox")).toBe(true);
  });
  it("has Attr Name", () => {
    expect(appRender.find(".checkbox").attr("name")).toEqual("hideBooks");
  });
  it("has Attr ID", () => {
    expect(appRender.find(".checkbox").attr("id")).toEqual("hideBooks");
  });
});
describe("CheckBox Component Snapshot", () => {
  it("Renders", () => {
    const component = renderer.create(
      <CheckBox
        _onChange={() => {}}
        label={"Hide Books"}
        name={"hideBooks"}
        id={"hideBooks"}
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
