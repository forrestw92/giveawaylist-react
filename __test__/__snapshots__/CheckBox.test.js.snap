// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox Component Snapshot Renders 1`] = `
<label
  htmlFor="hideBooks"
>
  <input
    className="checkbox"
    id="hideBooks"
    name="hideBooks"
    onChange={[Function]}
    type="checkbox"
  />
  Hide Books
</label>
`;
